plugin/plugin_test.go

```diff
@@ -0,0 +1,127 @@
+package plugin
+
+import (
+	context "context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"os"
+	"path/filepath"
+	"testing"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAeadPluginWrapper(t *testing.T) {
+	require := require.New(t)
+	ctx := context.Background()
+
+	pluginPath := os.Getenv("PLUGIN_PATH")
+	if pluginPath == "" {
+		t.Skipf("skipping plugin test as no PLUGIN_PATH specified")
+	}
+
+	wrapper, cleanup := TestPlugin(t, pluginPath)
```

Comment:
An alternate way to "execute" a plugin without having to compile one before testing would be to have the plugin's main func as a Test func, and make use of `-test.run=SomeWrapperMainFunc` to have go-plugin dispense a client whose command is an execution of the test func itself.

Vault does this via [TestAddTestPlugin](https://github.com/hashicorp/vault/blob/3ff381abe5aa894210146f01bfde362177fa00dd/vault/testing.go#L496). Sample usage can be found [here](https://github.com/hashicorp/vault/blob/3ff381abe5aa894210146f01bfde362177fa00dd/builtin/logical/database/backend_test.go#L49-L67).

Reply:
I prefer this method, because it aligns much more closely with what's actually going to happen.